elide-core/src/main/java/com/yahoo/elide/graphql/PersistentResourceFetcher.java

```diff
@@ -263,12 +266,382 @@ private Object fetchRelationship(Environment context) {
+        } else { /* must update object */
+            return updateObject(fetchEntries, input, fieldName, requestScope, parent);
+        }
+    }
+
+    /**
+     * Strips out the attributes from the {@code element} singleton list and returns just the relationships, if present.
+     * @param element data input
+     * @param entityClass entity class
+     * @param dictionary entity dictionary
+     * @return relationship map
+     */
+    private Map<String, Object> stripAttributes(List<Map<String, Object>> element, Class<?> entityClass,
+                                                EntityDictionary dictionary) {
+        Map<String, Object> relationshipsOnly = new HashMap<>();
+        Map<String, Object> entry = element.get(0); /* 'element' will always be singleton */
+
+        for(String key : entry.keySet()) {
+            if(dictionary.isRelation(entityClass, key)) {
+                relationshipsOnly.put(key, entry.get(key));
+            }
+        }
+        return relationshipsOnly;
+    }
+
+    /**
+     * Strips out the relationships from the {@code element} singleton list and returns just the attributes, if present.
+     * @param element data input
+     * @param entityClass entity class
+     * @param dictionary entity dictionary
+     * @return attributes map
+     */
+    private Map<String, Object> stripRelationships(List<Map<String, Object>> element, Class<?> entityClass,
+                                                   EntityDictionary dictionary) {
+        Map<String, Object> attributesOnly = new HashMap<>();
+        Map<String, Object> entry = element.get(0); /* 'element' will always have just one entry */
+
+        for(String key : entry.keySet()) {
+            if(dictionary.isAttribute(entityClass, key)) {
+                attributesOnly.put(key, entry.get(key));
+            }
+            if(Objects.equals(key, dictionary.getIdFieldName(entityClass))) {
+                attributesOnly.put(key, entry.get(key));
+            }
+        }
+        return attributesOnly;
+    }
+
+    /**
+     * Creates a new "root-level" resource in the database
+     * @param input individual input data objects
+     * @param outputType graphql-java output type
+     * @param fieldName graphql-java field type
+     * @param requestScope request scope
+     * @param uuid uuid
+     * @return {@link PersistentResource} object
+     */
+    private PersistentResource createObject(Map<String, Object> input, String fieldName,
+                                RequestScope requestScope, Optional<String> uuid,
+                                GraphQLType outputType, PersistentResource parent) {
+        EntityDictionary dictionary = requestScope.getDictionary();
+
+        Class<?> entityClass;
+        if(parent == null) {
+            GraphQLObjectType objectType = (GraphQLObjectType) ((GraphQLList) outputType).getWrappedType();
```

Comment:
Can we do the same thing we did for updateObject where we use the fieldName to extract the entityClass:
```
entityClass = dictionary.getEntityClass(fieldName);
```